python-cffi/netplan/netdef.py

```diff
@@ -75,8 +75,8 @@ def set_name(self) -> str:
         return _string_realloc_call_no_error(lambda b: lib.netplan_netdef_get_set_name(self._ptr, b, len(b)))
 
     @property
-    def keep_configuration(self) -> str:
-        return _string_realloc_call_no_error(lambda b: lib.netplan_netdef_get_keep_configuration(self._ptr, b, len(b)))
+    def critical(self) -> bool:
```

Comment:
I wonder if we could keep the Python binding untouched and return False if critical is false (or unset) and True if it's set to any of the other values...

Although we probably should change it to return the enum values. You can see an example of enums in the Python code here https://github.com/chr4/netplan/blob/feat/keep-configuration/python-cffi/netplan/_utils.py#L27

Reply:
That's an interesting thought. I feel like we should not diverge the bindings from the native API behavior, though. But we need to think about how we can handle that in a backwards compatible way.